xla/service/gpu/matmul_utils.cc

```diff
@@ -385,6 +385,8 @@ absl::StatusOr<bool> CanFoldTransposeOperandIntoDot(const HloInstruction& dot,
   switch (output_shape.element_type()) {
     case F8E4M3FN:
     case F8E5M2:
+    case F8E4M3FNUZ:
+    case F8E5M2FNUZ:
```

Comment:
This is actually different from the case in gemm_rewriter. It is determining whether the matmul is supported by XLA before the gemm rewrite. If I remove them here, it will result in an internal error: "Unexpected GEMM datatype:".

Reply:
@reedwm I have addressed most of the review comments. This one could not be removed as it would result in an error. Could you take a look at the changes and let me know if there are further concerns?